services/chaospod_test.go

```diff
@@ -0,0 +1,1465 @@
+					})
+
+				})
+
+				When("when the removeFinalizerForChaosPod return an error", func() {
+
+					BeforeEach(func() {
+						// Arrange
+						errorStatus := errors.StatusError{ErrStatus: metav1.Status{
+							Message: "node is not ready",
+						}}
+						targetSelectorMock.ExpectedCalls = nil
+						targetSelectorMock.EXPECT().TargetIsHealthy(mock.Anything, mock.Anything, mock.Anything).Return(&errorStatus)
+
+						k8sClientMock.EXPECT().Update(mock.Anything, mock.Anything).Return(fmt.Errorf("an error happened"))
+					})
+
+					It("should not remove the finalizer", func() {
+						// Assert
+						By("return an error")
+						Expect(err).Should(HaveOccurred())
+
+						By("not remove the finalizer")
+						Expect(isFinalizerRemoved).To(BeFalse())
+					})
+				})
+
+			})
+
+			Context("with a succeeded pod", func() {
+
+				BeforeEach(func() {
+					// Arrange
+					cpBuilder.WithStatusPhase(v1.PodSucceeded)
+				})
+
+				When("the k8s client return an error during the update", func() {
+
+					BeforeEach(func() {
+						// Arrange
+						k8sClientMock.EXPECT().Update(mock.Anything, mock.Anything, mock.Anything).Return(fmt.Errorf("could not update"))
+					})
+
+					It("should propagate the error", func() {
+						// Assert
+						Expect(err).Should(HaveOccurred())
+						Expect(isFinalizerRemoved).To(BeFalse())
+					})
+				})
+
+			})
+
+			Context("with a running node failure chaos pod ", func() {
+				BeforeEach(func() {
+					cpBuilder.WithStatusPhase(v1.PodRunning).WithChaosPodLabels(DefaultDisruptionName, DefaultNamespace, "", chaostypes.DisruptionKindNodeFailure)
+				})
+
+				When("the k8s client return an error during the update", func() {
+
+					BeforeEach(func() {
+						// Arrange
+						k8sClientMock.EXPECT().Update(mock.Anything, mock.Anything, mock.Anything).Return(fmt.Errorf("could not update"))
+					})
+
+					It("should propagate the error", func() {
+						// Assert
+						Expect(err).Should(HaveOccurred())
+						Expect(isFinalizerRemoved).To(BeFalse())
+					})
+				})
+			})
+		})
+
+		Context("with a chaos pod not being deleted", func() {
+
+			BeforeEach(func() {
+				// Arrange
+				chaosPod = builderstest.NewPodBuilder("test", DefaultNamespace).WithChaosPodLabels(DefaultDisruptionName, DefaultNamespace, "", "").WithChaosFinalizer().Build()
+			})
+
+			It("should not remove the finalizer", func() {
+				// Assert
+				By("not return an error")
+				Expect(err).ShouldNot(HaveOccurred())
+
+				By("not remove the finalizer")
+				k8sClientMock.AssertNotCalled(GinkgoT(), "Update", mock.Anything, mock.Anything)
+				Expect(isFinalizerRemoved).To(BeFalse())
+			})
+		})
+
+		Context("with a chaos pod without finalizer", func() {
+
+			BeforeEach(func() {
+				// Arrange
+				chaosPod = builderstest.NewPodBuilder("test", DefaultNamespace).WithChaosPodLabels(DefaultDisruptionName, DefaultNamespace, "", "").WithDeletion().Build()
+			})
+
+			It("should not remove the finalizer", func() {
```

Comment:
Is this what we want to happen? It would seem to me that `isFinalizerRemoved` is... true in a sense, as the finalizer has definitely been removed, and so we wouldn't want to mark this pod as stuck on removal

Reply:
It makes sense 🤔  I was more thinking, "the finalizer is already removed so we don't launch the action to remove it so we return false". I know, it is not really logic 😅